script.js

```diff
@@ -1,4 +1,254 @@
+/**
+ * scissors-paper-stone
+ */
+
+//The program internally randomly chooses 'scissors', 'paper', or 'stone'.
+//define function for generating a random integer
+var generateRandomInteger = function (max) {
+  return 1;
+  var randomDecimal = Math.random() * max;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNumber = randomInteger + 1;
+  console.log(randomNumber);
+  return randomNumber;
+};
+
+//assign the three objects to the randomly generated integers ranging from 1-3.
+var programObject = function () {
+  var totalObjects = 3;
+  var randomObjectNum = generateRandomInteger(totalObjects);
+
+  if (randomObjectNum == 1) {
+    console.log("paper");
+    return "paper";
+  } else if (randomObjectNum == 2) {
+    console.log("stone");
+    return "stone";
+  } else if (randomObjectNum == 3) {
+    console.log("scissors");
+    return "scissors";
+  }
+};
+
+//user inputs one of 'scissors', 'paper', or 'stone'.
+
+//the program outputs whether the user won, the program won, or it's a draw.
+
+//rules: scissors beats paper, paper beats stone, stone beats scissors, draw if both parties choose the same object.
+
+//rules for reversed SPS: reversed stone beats paper, reversed scissors beats stone, reversed paper beats scissors, draw if both parties choose the same object.
+
+var currentGameMode = "submit username";
+var userWon = 0;
+var computerWon = 0;
+var draw = 0;
+var turnsPlayed = userWon + computerWon + draw;
+var percentageWon = 0;
+var userName = "";
+var myOutputValue = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var program = programObject();
+
+  //when game mode is 'submit username', user has to enter username.
+  if (currentGameMode == "submit username") {
+    userName = input;
+    // after user enters username, game mode changes to 'scissors-paper-stone'
+    currentGameMode = "scissors-paper-stone";
+    myOutputValue = `Hello ${userName}! <br>Let's play scissors ✂️- paper 🗒- stone 🥌!<br>To play, enter scissors, paper, or stone, then click submit.`;
+  } else if (currentGameMode == "scissors-paper-stone") {
+    //format the input object name
+    var inputObjectFormatted = function (input) {
+      if (input == "scissors") {
+        return "scissors ✂️";
+      }
+      if (input == "paper") {
+        return "paper 🗒";
+      }
+      if (input == "stone") {
+        return "stone 🥌";
+      }
```

Comment:
I like your use of function to nicely format the output ✂️🗒🥌, although it would be clearer if this was placed with your other helper functions above the main function so that the main function is purely used for the main program